### ISSUE TITLE
indent conflicts with react/jsx-closing-bracket-location

### ISSUE THREAD

**Issue by jimmy-e** [eslint/eslint#11974]
**Tell us about your environment**

* **ESLint Version:** 5.16
* **Node Version:** 11.4.0
* **yarn Version:** 1.9.4

**What parser (default, Babel-ESLint, etc.) are you using?** `@tylescript-eslint/parser`

**Please show your full configuration:**

.eslintrc:
```
module.exports = {
  env: {
    'jest/globals': true
  },
  extends: [
    'airbnb',
    'plugin:@typescript-eslint/recommended',
  ],
  globals: {
    'document': true,
    'window': true,
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    project: './tsconfig.json',
  },
  plugins: ['@typescript-eslint', 'jest'],
  rules: {
    'indent': 'off',
    '@typescript-eslint/indent': [
      'error',
      2,
      {
        'ignoredNodes': ['JSXElement *'],
        'SwitchCase': 1,
      },
    ],
  },
  settings: {
    'import/extensions': [
      '.js',
      '.jsx'
    ],
    'import/resolver': {
      webpack: {
        config: 'webpack.common.js',
      }
    }
  },
};
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

1. Clone [this repo](https://github.com/jimmy-e/typescript-eslint-debug).
2. run `yarn`
3. run `yarn build:dev`

**What did you expect to happen?**

No errors. Aligning the closing bracket should not cause an indent error. I am stating that the indent rule should be ignored for JSX Elements. Note that indenting the closing bracket will conflict with the react/jsx-closing-bracket-location rule.

sampleComponent.jsx:
```
import React from 'react';

const SampleComponent = () => (
  <h
    custom-attr-one="1"
    custom-attr-two="2"
    custom-attr-three="3"
  > // <- causes failure
    custom header
  </h>
);

export default SampleComponent;
```

error:
```
ERROR in ./src/sampleComponent.jsx
Module build failed (from ./node_modules/eslint-loader/index.js):
Module failed because of a eslint error.

/Users/jemery/dev/typescript-eslint-debug/src/sampleComponent.jsx
  8:1  error  Expected indentation of 4 spaces but found 2  @typescript-eslint/indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

 @ ./src/index.jsx 4:0-48 5:27-42
 @ multi babel-polyfill ./src/index.jsx
error Command failed with exit code 2.
```

[ESLint Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogWydlcnJvcicsIDIsIHsgJ2lnbm9yZWROb2Rlcyc6IFsnSlNYRWxlbWVudCAqJ10sIH1dICovXG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuXG5jb25zdCBTYW1wbGVDb21wb25lbnQgPSAoKSA9PiAoXG4gIDxoXG4gICAgY3VzdG9tLWF0dHItb25lPVwiMVwiXG4gICAgY3VzdG9tLWF0dHItdHdvPVwiMlwiXG4gICAgY3VzdG9tLWF0dHItdGhyZWU9XCIzXCJcbiAgPlxuICAgIGN1c3RvbSBoZWFkZXJcbiAgPC9oPlxuKTtcblxuZXhwb3J0IGRlZmF1bHQgU2FtcGxlQ29tcG9uZW50OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnsiY29tbW9uanMiOmZhbHNlLCJub2RlIjpmYWxzZX19fQ==)

**Are you willing to submit a pull request to fix this bug?**
Yes

**Comment by platinumazure** [comment]
Hi @jimmy-e, thanks for the issue.

You've listed a custom parser and, in fact, a custom rule. Can you please confirm that this issue also occurs when using the default parser and the core `indent` rule?

**Comment by jimmy-e** [comment]
Thanks @platinumazure.

[The repo has been updated.](https://github.com/jimmy-e/typescript-eslint-debug)

**.estlintrc:**
```js
module.exports = {
  env: {
    'jest/globals': true
  },
  extends: 'airbnb',
  globals: {
    'document': true,
    'window': true,
  },
  parser: 'babel-eslint',
  plugins: ['jest'],
  rules: {
    indent: ['error', 2, { 'ignoredNodes': ['JSXElement *'], }]
  },
  settings: {
    'import/extensions': [
      '.js',
      '.jsx'
    ],
    'import/resolver': {
      webpack: {
        config: 'webpack.common.js',
      }
    }
  },
};
```

**.sampleComponent.jsx:**
```js
import React from 'react';

const SampleComponent = () => (
  <h
    custom-attr-one="1"
    custom-attr-two="2"
    custom-attr-three="3"
  >
    custom header
  </h>
);

export default SampleComponent;
```

**.sampleComponentTwo.jsx:**
```js
import React from 'react';

const SampleComponentTwo = () => (
  <h
    custom-attr-one="1"
    custom-attr-two="2"
    custom-attr-three="3"
    >
    custom header
  </h>
);

export default SampleComponentTwo;
```

**error:**
```
ERROR in ./src/sampleComponent.jsx
Module build failed (from ./node_modules/eslint-loader/index.js):
Module failed because of a eslint error.

/Users/jemery/dev/typescript-eslint-debug/src/sampleComponent.jsx
  8:1  error  Expected indentation of 4 spaces but found 2  indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

 @ ./src/index.jsx 4:0-48 6:69-84
 @ multi babel-polyfill ./src/index.jsx

ERROR in ./src/sampleComponentTwo.jsx
Module build failed (from ./node_modules/eslint-loader/index.js):
Module failed because of a eslint error.

/Users/jemery/dev/typescript-eslint-debug/src/sampleComponentTwo.jsx
  8:5  error  The closing bracket must be aligned with the line containing the opening tag (expected column 3)  react/jsx-closing-bracket-location

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

 @ ./src/index.jsx 5:0-54 6:113-131
 @ multi babel-polyfill ./src/index.jsx
error Command failed with exit code 2.
```

**Comment by platinumazure** [comment]
What happens if you configure your ignored nodes to include JSXElement as well (not just `JSXElement *`)?

**Comment by OliverSieweke** [comment]
I've just encountered the same issue and including `JSXElement` in the `ignoredNodes` of the `indent` rule fixed it for me.

**Comment by jimmy-e** [comment]
Yep that works, thanks!

**Comment by andrewjrhill** [comment]
As suggested by OliverSieweke, this following `"indent"` configuration is working for me:

```JSON
"rules": {
    "indent": ["error", 4, {"ignoredNodes": ["JSXElement", "JSXElement *"]}],
}
```
